src/transactions/TransactionFrame.cpp

```diff
@@ -286,6 +286,46 @@ TransactionFrame::processSeqNum(LedgerManager& lm, LedgerDelta& delta)
     }
 }
 
+bool
+TransactionFrame::processSignatures(SignatureChecker& signatureChecker,
+                                    Application& app, LedgerDelta& delta)
+{
+    if (app.getLedgerManager().getCurrentLedgerVersion() < 10)
+    {
+        return true;
+    }
+
+    auto allOpsValid = std::all_of(
```

Comment:
this implementation has a problem (compared with the version you had before that was a loop):
if the one time signature is used in an operation after a failed auth operation, it will not be marked as "used", and will stick around (making `removeUsedOneTimeSignerKeys` no-op).

In order to preserve that property, you could use instead `std::accumulate` but given what we're doing, it's actually shorter and more readable to just use a for loop like before.